### PR TITLE
Rework parameters sidebar to mantine

### LINKED ISSUE #41140: Rework filter settings sidebar to mantine

### PR DESCRIPTION
**Pull request by uladzimirdev**
Closes https://github.com/metabase/metabase/issues/41140

### Description

Rework Parameters sidebar to mantine:
- use tabs
- replace Toggle with Switch
- replace some styled components elements with layout components from mantine

### How to verify

check parameter settings sidebar

### Demo

https://github.com/metabase/metabase/assets/125459446/009d414c-d05a-4b09-adeb-bc2f6344164f

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.tsx
Comment (ranquild) [inline frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.tsx:165]: Please don't override it like that, it won't work in v7. I'd suggest leaving it as it is or contacting the design team if we need to change the component

**Review by ranquild — COMMENTED** [review]
@uladzimirdev everything is great except Tab style overrides. Please take a look at the comment

File: frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.tsx
Comment (uladzimirdev) [reply frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.tsx:165]: image

as I understood from https://mantine.dev/guides/6x-to-7x/#styles-prop only nesting of selectors doesn't work, so I presume mantine will not expand `&` with a className in `styles` prop, but it will still work.

I can try to rework it with class if you're sure that `styles` will not work in v7, but we already have several places using styles anyway https://github.com/search?q=repo%3Ametabase%2Fmetabase+styles%3D%7B&type=code

Comment (npretto) [reply frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.tsx:165]: Are we sure that's not going to work?
https://mantine.dev/guides/6x-to-7x/#styles-prop
Says that `nested selectors` won't work, but I think what they mean is something like 
image
They say that "Regular selectors are still supported":
image

Comment (uladzimirdev) [reply frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.tsx:165]: okay, waiting for design feedback https://metaboat.slack.com/archives/C02H619CJ8K/p1712668558824799

**Comment by uladzimirdev** [comment]
@ranquild I replaced styles with classnames ece7bfc769c8769d9c74ee8c1a088f6cc45b0c8d

if designers decide to override border-radius for all the components, I'll create a separate PR

**Comment by ranquild** [comment]
@uladzimirdev so far this is your sole decision to hack the component and add inconsistency. Why you are doing this?

**Comment by uladzimirdev** [comment]
> @uladzimirdev so far this is your sole decision to hack the component and add inconsistency. Why you are doing this?

well, it's clear, because otherwise component looks weird

**Comment by uladzimirdev** [comment]
I got design approval [here](https://metaboat.slack.com/archives/C0645JP1W81/p1712700328844099?thread_ts=1712610386.712429&channel=C0645JP1W81&message_ts=1712700328.844099)

File: frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.tsx
Comment (kamilmielnik) [inline frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.tsx:142]: IMO typing ids of tabs is an overkill and adds unnecessary complexity, which shows e.g. here.

I'd just go with `string`.

**Review by kamilmielnik — APPROVED** [review]
Visually it looks so much better :heart: 

Please reconsider the types for tab ids.
Other than that PR looks good, approving :+1:

File: frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.tsx
Comment (uladzimirdev) [reply frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.tsx:142]: I agree, at the same time it's so easy to make a mistake with plain strings, so I decided to type them. Having enum would make things even more complicated

**Pull request merged**
